proxy/ControlBase.cc

```diff
@@ -796,58 +795,120 @@ ControlBase::ProcessModifiers(matcher_line *line_info)
+      }
+
+      if (strcasecmp(label, "port") == 0) {
+        mod = PortMod::make(value, &errBuf);
+      } else if (strcasecmp(label, "iport") == 0) {
+        mod = IPortMod::make(value, &errBuf);
+      } else if (strcasecmp(label, "scheme") == 0) {
+        mod = SchemeMod::make(value, &errBuf);
+      } else if (strcasecmp(label, "method") == 0) {
+        mod = MethodMod::make(value, &errBuf);
+      } else if (strcasecmp(label, "prefix") == 0) {
+        mod = PrefixMod::make(value, &errBuf);
+      } else if (strcasecmp(label, "suffix") == 0) {
+        mod = SuffixMod::make(value, &errBuf);
+      } else if (strcasecmp(label, "src_ip") == 0) {
+        mod = SrcIPMod::make(value, &errBuf);
+      } else if (strcasecmp(label, "time") == 0) {
+        mod = TimeMod::make(value, &errBuf);
+      } else if (strcasecmp(label, "tag") == 0) {
+        mod = TagMod::make(value, &errBuf);
+      } else if (strcasecmp(label, "internal") == 0) {
+        mod = InternalMod::make(value, &errBuf);
+      } else {
+        err = ME_BAD_MOD;
+      }
+
+      if (errBuf) {
+        err = ME_CALLEE_GENERATED; // Mod make failed.
+      }
+
+      // If nothing went wrong, add the mod and bump the element count.
+      if (ME_UNKNOWN == err) {
+        _mods.push_back(mod);
+        --n_elts;
+      } else {
+        delete mod; // we still need to clean up because we didn't transfer ownership.
+      }
     }
-    if (!value) {
-      err = ME_PARSE_FAILED;
-      break;
-    }
-
-    if (strcasecmp(label, "port") == 0) {
-      mod = PortMod::make(value, &errBuf);
-    } else if (strcasecmp(label, "iport") == 0) {
-      mod = IPortMod::make(value, &errBuf);
-    } else if (strcasecmp(label, "scheme") == 0) {
-      mod = SchemeMod::make(value, &errBuf);
-    } else if (strcasecmp(label, "method") == 0) {
-      mod = MethodMod::make(value, &errBuf);
-    } else if (strcasecmp(label, "prefix") == 0) {
-      mod = PrefixMod::make(value, &errBuf);
-    } else if (strcasecmp(label, "suffix") == 0) {
-      mod = SuffixMod::make(value, &errBuf);
-    } else if (strcasecmp(label, "src_ip") == 0) {
-      mod = SrcIPMod::make(value, &errBuf);
-    } else if (strcasecmp(label, "time") == 0) {
-      mod = TimeMod::make(value, &errBuf);
-    } else if (strcasecmp(label, "tag") == 0) {
-      mod = TagMod::make(value, &errBuf);
-    } else if (strcasecmp(label, "internal") == 0) {
-      mod = InternalMod::make(value, &errBuf);
-    } else {
-      err = ME_BAD_MOD;
-    }
-
-    if (errBuf) {
-      err = ME_CALLEE_GENERATED; // Mod make failed.
-    }
-
-    // If nothing went wrong, add the mod and bump the element count.
-    if (ME_UNKNOWN == err) {
-      _mods.push_back(mod);
-      --n_elts;
-    } else {
-      delete mod; // we still need to clean up because we didn't transfer ownership.
+  } else { // parsing a yaml config
+    const char *labels[10] = {"port", "iport", "scheme", "method", "prefix", "suffix", "src_ip", "time", "tag", "internal"};
```

Comment:
Don't use magic constants! Make them symbolic constants. E.g.
```
static constexpr unsigned N_LABEL = 10;
std::array<const char*, N_LABEL> labels = { ....};
```

Reply:
made the suggested change.